sdk/monitor/monitor-opentelemetry/test/utils/assert.ts

```diff
@@ -19,11 +19,21 @@ export const assertData = (actual: MonitorBase, expected: MonitorBase): void =>
   assert.ok(actual.baseData);
   for (const [key, value] of Object.entries(expected.baseData!)) {
     const serializedKey = (EnvelopeMapper.type as any).modelProperties![key]?.serializedName ?? key;
-    assert.deepStrictEqual(
-      actual.baseData[serializedKey],
-      value,
-      `baseData.${serializedKey} should be equal\nActual: ${actual.baseData[serializedKey]}\nExpected: ${value}`,
-    );
+    if (typeof value === "object") {
+      for (const [nestedKey, nestedValue] of Object.entries(value)) {
+        assert.deepStrictEqual(
+          expected.baseData![serializedKey][nestedKey],
```

Comment:
Should we fail on null checks instead of just forcing to be `baseData!`

Reply:
Updated to assert that `expected.baseData` is defined.